src/Raven.Studio/typescript/widgets/virtualGrid/VirtualGrid.ts

```diff
@@ -0,0 +1,382 @@
+﻿/// <reference path="../../../typings/tsd.d.ts"/>
+
+import VirtualRow = require("widgets/virtualGrid/VirtualRow");
+import PagedResult = require("widgets/virtualGrid/PagedResult");
+import ItemFetch = require("widgets/virtualGrid/ItemFetch");
+import Column = require("widgets/virtualGrid/Column");
+import TextCellTemplate = require("widgets/virtualGrid/TextCellTemplate");
+import CheckedCellTemplate = require("widgets/virtualGrid/CheckedCellTemplate");
+import VirtualGridConfig = require("widgets/virtualGrid/VirtualGridConfig");
+
+class VirtualGrid {
+    items: Object[] = []; // The items loaded asynchronously.
+    totalItemCount: number | null = null;
+    virtualRows: VirtualRow[] = []; // These are the fixed number of elements that get displayed on screen. Each virtual row displays an element from .items array. As the user scrolls, rows will be recycled to represent different items.
+    fetcher: (skip: number, take: number) => JQueryPromise<PagedResult>; // The function invoked that fetches a chunk of items.
+    gridId: string;
+    gridElement: JQuery;
```

Comment:
in 4.0 we have convention to start jquery selector property with $, so it should be $gridElement